### ISSUE TITLE
Feature request: replace Android http module with OkHttp

### ISSUE THREAD

**Issue by alexrainman** [NativeScript/NativeScript#923]
OkHttp is way faster. It's like the counterpart of NSUrlSession for Android. Take a look a ModernHttpClient plugin for Xamarin.

**Comment by enchev** [comment]
Hey @alexrainman,

Nice suggestion! We evaluated actually OkHttp vs. our home-grown async http client for Android and the only concern we have is that this will be another ~300K to the app.

Let us know what you think!

**Comment by x4080** [comment]
Maybe as a plugin?

**Comment by alexrainman** [comment]
I know you may need to add OkHttp library and that will add extra size to the app, but, as soon as my http calls gets way faster, i'm ok with it ;)

OkHttp is really popular. As a plugin? I am ok with it too. Cnce i add it to the project, it needs to override the home-grown one and i will keep writing only one http code.

Good job guys!

**Comment by atanasovg** [comment]
OkHttp depends on the okio library. It adds additional ~80K, so the package grows to ~400K. But the size is not our only concern. The library itself adds many other features - like file-system caching - which may not be needed by a simple app that needs a simple http request. Probably the plugin approach would be indeed the better solution.

**Comment by alexrainman** [comment]
I hope to be testing that plugin as soon as possible. Can you point me to the right place to learn about async/await support in {N}?

**Comment by alexrainman** [comment]
Is your in-house http client for Android based on one of these?

java.net.HttpURLConnection
java.net.ssl.HttpsURLConnection
Android HttpClient
DefaultHTTPClient, using the standard Apache HttpClient library with the org.apache.http.package

I am analyzing the option to use NativeScript for enterprise and one the of requirements is Mobile Iron support. That's why i asked about OkHttp because Mobile Iron is working to support it.

And please, guide me where to read about async/await with {N} http.

**Comment by atanasovg** [comment]
We are using `URL.openConnection()` method, which will return an `HttpsURLConnection` if the requested url has a `https` scheme. Still, we do not apply any further additional settings to an Https connection. The actual android code is located here:
https://github.com/NativeScript/android-runtime/blob/master/src/src/com/tns/Async.java#L271

**Comment by alexrainman** [comment]
Great!

**Comment by enchev** [comment]
Hey @alexrainman,

Our own @kiril-stanoev made [this plugin](https://www.npmjs.com/package/nativescript-okhttp). Please try it and let us know if this will work for you!

Thanks

**Comment by alexrainman** [comment]
Awesome. I will give a try as soon as i can. Something i will love to see in your roadmap is the option to install this plugin overriding you default http client for android so i don't need to use OkHttp directly, instead i will love to write my code using the http module, got it? Thanks a lot for listening.

**Comment by maknz** [comment]
:+1:, we'd love to use OkHttp, but we wanna keep using the overall `http` cross-platform interface.

**Comment by alexrainman** [comment]
Exactly. No matter if OkHttp is optional plugin, once installed "we wanna keep using the overall http cross-platform interface".

**Comment by alexrainman** [comment]
Any news on this?

**Comment by jerbob92** [comment]
I created [a plugin](https://github.com/klippa-app/nativescript-http) that is what we think the core http should be. It's actually drop-in as it works exactly the same as the core http, we also added a Webpack plugin that alters the core http to make sure all http requests go through this plugin.

It currently only has a different implementation for Android (iOS works fine, but the same as in core http), I will make the implementation as soon as I have access to OSX (Corona struggles).

I'd love to hear some feedback.

**Comment by CatchABus** [comment]
Closing as not planned for now.
There are community plugins that make use of OkHttp and everyone is free to use them as an alternative.
